app/src/main/java/de/tum/in/tumcampusapp/adapters/BarrierfreeContactAdapter.java

```diff
@@ -0,0 +1,102 @@
+package de.tum.in.tumcampusapp.adapters;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.util.Linkify;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.List;
+
+import de.tum.in.tumcampusapp.R;
+import de.tum.in.tumcampusapp.activities.PersonsDetailsActivity;
+import de.tum.in.tumcampusapp.models.tumcabe.BarrierfreeContact;
+import de.tum.in.tumcampusapp.models.tumo.Person;
+import se.emilsjolander.stickylistheaders.StickyListHeadersAdapter;
+
+/**
+ * An adapter used to display contact information in barrierfree page.
+ */
+public class BarrierfreeContactAdapter extends SimpleStickyListHeadersAdapter<BarrierfreeContact>
+        implements StickyListHeadersAdapter {
+    public BarrierfreeContactAdapter(Context context, List<BarrierfreeContact> infos) {
+        super(context, infos);
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        final ViewHolder holder;
+        View view = convertView;
+
+        if(view == null){
+            view = getInflater().inflate(R.layout.activity_barrier_free_contact_listview, parent, false);
+
+            // Crate UI element
+            holder = new ViewHolder();
+            holder.name = (TextView) view
+                    .findViewById(R.id.barrierfreeContactListViewName);
+            holder.phone = (TextView) view
+                    .findViewById(R.id.barrierfreeContactListViewPhone);
+            holder.email = (TextView) view
+                    .findViewById(R.id.barrierfreeContactListViewEmail);
+            holder.more = (TextView) view
+                    .findViewById(R.id.barrierfreeContactListViewTumOnlineMore);
+
+            view.setTag(holder);
+        } else {
+            holder = (ViewHolder) view.getTag();
+        }
+
+        // display information of current person
+        final BarrierfreeContact contact = getInfoList().get(position);
+
+        if(!contact.ifValid()) {
+            view.setVisibility(View.GONE);
+            return view;
+        }
+
+        view.setVisibility(View.VISIBLE);
+
+        holder.name.setText(contact.getName());
+
+        holder.phone.setText(contact.getTelephone(), TextView.BufferType.SPANNABLE);
+        Linkify.addLinks(holder.phone, Linkify.ALL);
+
+        holder.email.setText(contact.getEmail());
```

Comment:
Could you please make this method a bit more concise by extracting methods for setting up the ViewHolder as methods of the ViewHolder?